internal/push/check.go

```diff
@@ -93,13 +92,13 @@ func (c *Check) executeChecker() {
 
 	// Do not run checks against hostnames that resolve to private IPs unless allowed.
 	if ptrToBool(c.config.AllowPrivateIPs) || helpers.IsScannable(c.config.Check.Target) {
-		err = c.checker.Run(c.ctx, c.config.Check.Target, c.config.Check.Opts, runtimeCheckState)
+		err = c.checker.Run(c.ctx, c.config.Check.Target, c.config.Check.AssetType, c.config.Check.Opts, runtimeCheckState)
 		// We always execute the cleanup function after the check has finished.
 		// We use a fresh new context because here the origin context created for
 		// running the check can be finalized.
-		c.checker.CleanUp(context.Background(), c.config.Check.Target, c.config.Check.Opts)
+		c.checker.CleanUp(context.Background(), c.config.Check.Target, c.config.Check.AssetType, c.config.Check.Opts)
 	} else {
-		err = fmt.Errorf("target is not scannable")
+		err = state.ErrAssetUnreachable
```

Comment:
I know I proposed this, but one question: the inconclusive state is retro-compatible?

Reply:
So, after testing again there is a misbehave with the current scan-engine version when it receives the `INCONCLUSIVE` state because it's not a terminal state for it. The whole flow would be compatible if vulcan-agent was also running on "previous version" (before including awareness of the INCONCLUSIVE state), because then any error would produce a FAILED state, which is recognized by the scan-engine.

So all in all, and considering our situation, I think best option is to skip reachability verification when asset type is not set.

Reply:
👍